File: znld-web/src/main/java/com/sybd/znld/controller/device/dto/ExecuteResult.java
package com.sybd.znld.controller.device.dto;

import com.sybd.znld.core.BaseApiResult;
import com.sybd.znld.onenet.dto.OneNetExecuteResult;
import io.swagger.annotations.ApiModel;

@ApiModel(value = "执行命令后的返回值")
public class ExecuteResult extends BaseApiResult{
    public ExecuteResult(){}
    public ExecuteResult(Integer code, String msg){
        super(code, msg);
    }
    public static ExecuteResult fail(String msg){
        return new ExecuteResult(1, msg);
    }
    public static ExecuteResult success(OneNetExecuteResult result){
        return new ExecuteResult(0, "");
    }
}

File: znld-web/src/main/java/com/sybd/znld/controller/device/dto/ExecuteResult.kt
package com.sybd.znld.controller.device.dto;

import com.sybd.znld.core.BaseApiResult;
import com.sybd.znld.onenet.dto.OneNetExecuteResult
import io.swagger.annotations.ApiModel;

@ApiModel(value = "执行命令后的返回值")
class ExecuteResult(code: Int, msg: String): BaseApiResult(code, msg) {
    companion object {
        @JvmStatic fun fail(msg: String): ExecuteResult{
            return ExecuteResult(1, msg)
        }
        @JvmStatic fun success(result: OneNetExecuteResult): ExecuteResult{
            return ExecuteResult(0, "")
        }
    }
}
